lib/Image/Adapter/Imagick.php

```diff
@@ -985,6 +983,40 @@ public function supportsFormat(string $format, bool $force = false): bool
         return self::$supportedFormatsCache[$format];
     }
 
+    /**
+     * @return array{int, int}
+     */
+    private function calculateOverlayPosition(\Imagick $newImage, int $x, int $y, string $origin): array
+    {
+        $imageWidth = $this->resource->getImageWidth();
+        $imageHeight = $this->resource->getImageHeight();
+        $newImageWidth = $newImage->getImageWidth();
+        $newImageHeight = $newImage->getImageHeight();
+
+        switch ($origin) {
+            case 'top-right':
+                $x = $imageWidth - $newImageWidth - $x;
+
+                break;
+            case 'bottom-left':
+                $y = $imageHeight - $newImageHeight - $y;
+
+                break;
+            case 'bottom-right':
+                $x = $imageWidth - $newImageWidth - $x;
+                $y = $imageHeight - $newImageHeight - $y;
+
+                break;
+            case 'center':
+                $x = round($imageWidth / 2 - $newImageWidth / 2) + $x;
+                $y = round($imageHeight / 2 - $newImageHeight / 2) + $y;
+
+                break;
+        }
+
+        return [(int)$x, (int)$y];
```

Comment:
What do you think of this?

```suggestion
        return match ($origin) {
            'top-right' => [
                $imageWidth - $newImageWidth - $x,
                $y,
            ],
            'bottom-left' => [
                $x,
                $imageHeight - $newImageHeight - $y,
            ],
            'bottom-right' => [
                $imageWidth - $newImageWidth - $x,
                $imageHeight - $newImageHeight - $y,
            ],
            'center' => [
                (int) round($imageWidth / 2 - $newImageWidth / 2) + $x,
                (int) round($imageHeight / 2 - $newImageHeight / 2) + $y,
            ],
            default => [
                $x,
                $y,
            ],
        };
```